app.js

```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const app = express();
+const { getTopics } = require('./controllers/topicsController');
+const db = require('./db'); // Import the database pool
+
+app.use(express.json());
+
+app.get('/api/topics', getTopics);
+
+app.use((err, req, res, next) => {
+  console.error(err.stack);
+  res.status(500).json({ error: "internal server error" });
+});
+
+app.listen(4000, () => {
```

Comment:
We really want to extract our listen file, due to the fact that supertest wants to inject its own listener